packages/design-system/src/components/IdleTimeout/IdleTimeout.tsx

```diff
@@ -77,70 +78,148 @@ const defaultMessageFormatter = (timeTilTimeout: number): React.ReactNode => {
   );
 };
 
+// names for local storage variables
+const timeoutCookieName = 'CMS_DS_TIMEOUT';
+const timeoutWarningCookieName = 'CMS_DS_WARNING';
+
 const IdleTimeout = ({
-  closeDialogText = 'Close',
   continueSessionText = 'Continue session',
   heading = 'Are you still there?',
   endSessionButtonText = 'Logout',
+  endSessionRedirectUrl = '/logout',
   formatMessage = defaultMessageFormatter,
-  onClose,
   onSessionContinue,
   onSessionForcedEnd,
   onTimeout,
   showSessionEndButton = false,
   timeToTimeout,
   timeToWarning = 5,
-}: IdleTimeoutProps) => {
-  const handleContinueSession = () => {
-    // reset countdown timer
-    if (onSessionContinue) {
-      onSessionContinue();
+}: IdleTimeoutProps): JSX.Element => {
+  if (timeToWarning > timeToTimeout) {
+    console.error(
+      'Error in TimeoutManager component. `timeToWarning` is greater or equal to `timeToTimeout`'
+    );
+  }
+  // convert minutes to milliseconds
+  const msToTimeout = timeToTimeout * 60000;
+  const msToWarning = (timeToTimeout - timeToWarning) * 60000;
+  const [checkStatusTime, setCheckStatusTime] = useState<number>(null);
```

Comment:
Why does this value to be saved in local state?

Reply:
I wanted to clear the checkStatus interval when the timeout occurs and using the custom hook, that happens by changing the interval time from a number to null.

Reply:
👍 and it's being cleared before component destruction so that if app devs want to show some other messaging this modal doesn't pop up again? It would only be useful for unauthentiicated apps, right? I actually think in the example front-end-only app that we've been looking at, it re-routes to the landing page that doesn't render the idle timeout (because the end user hasn't entered any sensitive information yet). Maybe we can actually just let this clear on component destruction like `useInterval` does and leave the ball in product teams' court to handle what happens after a timeout.